infra/base-images/base-builder/detect_repo.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module to get the the name of a git repo containing a specific commit
+inside of an OSS-Fuzz project.
+
+Example Usage:
+
+  detect_repo.py --src_dir /src --example_commit b534f03eecd8a109db2b085ab24d419b6486de97
+
+Prints the location of the git remote repo as well as the repos name seperated by a space.
+
+  https://github.com/VirusTotal/yara.git yara
+
+"""
+import argparse
+import os
+import subprocess
+
+
+def main():
+  """Function to get a git repos information based on its commit."""
+  parser = argparse.ArgumentParser(
+      description='Finds a specific git repo in an oss-fuzz projects docker file.'
+  )
+  parser.add_argument(
+      '--src_dir',
+      help='The location of th oss-fuzz projects source directory',
+      required=True)
+  parser.add_argument(
+      '--example_commit',
+      help='A commit SHA refrencing the projects main repo',
+      required=True)
+  args = parser.parse_args()
+  out, _ = run_command(['ls'], location=args.src_dir, check_result=True)
+  out = out.replace('\n', ' ')
+  for single_dir in out.split(' '):
```

Comment:
we can do `os.listdir(args.src_dir)` instead of doing `ls` and splitting the output manually

Reply:
things like `rstrip()` also won't be needed then

Reply:
Done